blockchain/blockchain.go

```diff
@@ -507,9 +535,53 @@ func (b *Blockchain) Finalise(
 			return err
 		}
 
-		// Update chain height
+		block, err := core.GetBlockByNumber(txn, block.Number)
+		if err != nil {
+			return err
+		}
+		su, err := core.GetStateUpdateByBlockNum(txn, block.Number)
+		if err != nil {
+			return err
+		}
+		newBlock = block
+		newSU = su
+		return nil
+	}
+
+	// Simulate without commit
+	txn := b.database.NewIndexedBatch()
+	if err := simulate(txn); err != nil {
+		return nil, nil, nil, nil, err
+	}
+	txn.Reset()
+	return newBlock, newSU, newCommitments, concatCount, nil
+}
+
+// Finalise appends the block to the chain, assuming the checks pass
```

Comment:
I would suggest:
```suggestion
// Finalise checks the block correctness and appends it to the chain
```
Or something similar, I think it's easier to read